File: backend/app/api/company.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from app.schemas.company import CompanyCreate, CompanyUpdate, CompanyInDB
from app.database import get_db
from app.core import crud_company

router = APIRouter()


@router.get("/suggest", response_model=list[CompanyInDB])
def suggest_companies(keyword: str = Query(..., min_length=1), db: Session = Depends(get_db)):
    return crud_company.suggest_companies(db, keyword)


@router.get("/{company_id}", response_model=CompanyInDB)
def get_company(company_id: int, db: Session = Depends(get_db)):
    company = crud_company.get_company(db, company_id)
    if not company:
        raise HTTPException(status_code=404, detail="Company not found")
    return company


@router.post("/create", response_model=CompanyInDB)
def create_company(data: CompanyCreate, db: Session = Depends(get_db)):
    # 在创建之前检查公司是否已存在
    existing_company = crud_company.get_company_by_name(db, data.name)
    if existing_company:
        # 如果公司已存在，选择更新
        updated_company = crud_company.update_company(
            db, existing_company.id, data)
        return updated_company
    else:
        # 如果公司不存在，执行创建操作
        return crud_company.create_company(db, data)


@router.put("/{company_id}", response_model=CompanyInDB)
def update_company(company_id: int, data: CompanyUpdate, db: Session = Depends(get_db)):
    company = crud_company.update_company(db, company_id, data)
    if not company:
        raise HTTPException(status_code=404, detail="Company not found")
    return company

File: frontend/src/api/company.js
import axios from 'axios';  // 直接用axios

const url = import.meta.env.VITE_API_BASE_URL
const baseURL = ` ${url}/api/company`


// 搜索公司（模糊匹配）
export function suggestCompanies(keyword) {
    return axios.get(`${baseURL}/suggest`, {
        params: { keyword }
    });
}

// 根据ID获取公司详情
export function getCompany(companyId) {
    return axios.get(`${baseURL}/${companyId}`);
}

// 创建公司
export function createCompany(data) {
    return axios.post(`${baseURL}/create`, data);
}

// 更新公司
export function updateCompany(companyId, data) {
    return axios.put(`${baseURL}/${companyId}`, data);
}
